src/ui/Toggle/ToggleUI.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useState } from 'react';
+import { useToggleContext } from './ToggleContext';
+import { filterNumber } from './utils';
+
+export interface ToggleUIProps {
+  reversed?: boolean;
+  width?: string;
+  animationDuration?: string;
+  style?: Record<string, string>;
+  name?: string;
+  id?: string;
+  ariaLabel?: string;
+  ariaLabelledby?: string;
+}
+
+export function ToggleUI(props: ToggleUIProps): React.ReactElement {
+  const {
+    reversed = false,
+    width = '40px',
+    animationDuration = '0.5s',
+    style = {},
+    name = '',
+    id = '',
+    ariaLabel = '',
+    ariaLabelledby = '',
+  } = props;
+  const {
+    checked,
+    disabled,
+    onChange,
+    onFocus,
+    onBlur,
+  } = useToggleContext();
+
+  // animation should not be activated in the initialization step
+  const [animatedClassName, setAnimated] = useState('');
```

Comment:
```suggestion
  const [animatedClassName, setAnimatedClassName] = useState('');
```
Why don't we give the exact name for the dispatcher?